### PR TITLE
Feat/sentry monitoring

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Implemented Sentry monitoring for the application in production environment

## Explain why these changes are made
We want to track errors that occurs in the application in our production environment

## Explain your solution
Created a factory which returns a sentry service while in production. This service will then communicate with our provided sentry server which is going to be added as an environment variable

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. In development it should run as usual
4. If testing app in production (APP_ENV = production), then ask @kajm for credentials

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [x] Building the Application on a Android device/simulator.

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1v2ue1v Koppla appen till Sentry](https://app.clickup.com/t/1v2ue1v)

File: source/services/MonitoringService.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  init: () => void;
+  sendError: (Error: Error | string | null) => void;
+  wrap: (component: ComponentType<unknown>) => ComponentType<unknown>;
+}
+
+const nullMonitoringService: MonitoringService = {
+  init: () => undefined,
+  sendError: () => undefined,
+  wrap: (component) => component,
+};
```
Comment (MasterKenth) [inline source/services/MonitoringService.ts:16]: Could we move `nullMonitoringService` to a separate file? I think it's more concise to have concrete implementations in separate files even if it's a "fallback" implementation.

File: source/services/MonitoringService.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class MonitoringServiceFactory {
+  static getMonitoringService(environment: string): MonitoringService {
+    if (environment === "production") {
```
Comment (MasterKenth) [inline source/services/MonitoringService.ts:20]: Do we have constants for `"production"` and `"development"`?

File: source/services/MonitoringService.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      return SentryMonitoringService;
+    }
+
+    return nullMonitoringService;
+  }
+}
+
+export function getMonitoringService(): MonitoringService {
+  return MonitoringServiceFactory.getMonitoringService(
+    (env.APP_ENV ?? "develop") as string
```
Comment (MasterKenth) [inline source/services/MonitoringService.ts:30]: `"develop"` should be `"development"`.

File: source/services/MonitoringService.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ComponentType } from "react";
+import env from "react-native-config";
+
+import SentryMonitoringService from "./SentryMonitoringService";
+
+export interface MonitoringService {
+  init: () => void;
+  sendError: (Error: Error | string | null) => void;
+  wrap: (component: ComponentType<unknown>) => ComponentType<unknown>;
+}
```
Comment (MasterKenth) [inline source/services/MonitoringService.ts:10]: _Maybe_ move interface to a separate file, to separate definitions from declarations (might be overkill here).

File: source/services/SentryMonitoringService.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import * as Sentry from "@sentry/react-native";
+import env from "react-native-config";
+
+import type { MonitoringService } from "./MonitoringService";
+
+const sentryMonitoringService: MonitoringService = {
+  init: () => {
+    Sentry.init({
+      dsn: env.SENTRY_DSN,
+      // Set tracesSampleRate to 1.0 to capture 100% of transactions for performance monitoring.
```
Comment (MasterKenth) [inline source/services/SentryMonitoringService.ts:10]: Comment necessary? Doesn't seem to add much value.

**Review by MasterKenth — CHANGES_REQUESTED** [review]

File: source/helpers/error-handler/ErrorHandler.tsx [hunk 6]
```diff
@@ -15,23 +16,34 @@ import { string } from 'prop-types';
...
+      Alert.alert("Something went wrong in storybook", `${errorMessage}`, [
+        { text: "OK" },
+      ]);
     } else {
-      Alert.alert('Something went wrong in Mitt Helsingborg', `${errorMessage}`, [
-        {
-          text: 'Restart',
-          onPress: () => {
-            RNRestart.Restart();
+      getMonitoringService().sendError(error);
```
Comment (MasterKenth) [inline source/helpers/error-handler/ErrorHandler.tsx:34]: Maybe we should handle non-fatal errors as well? Those are `console.error(...)` whereas fatal are uncaught exceptions.

File: source/services/MonitoringService.ts (hunk 1 above)
Comment (kajm) [reply source/services/MonitoringService.ts:16]: Fixed

File: source/services/MonitoringService.ts (hunk 2 above)
Comment (kajm) [reply source/services/MonitoringService.ts:20]: Doesn't seems like it

File: source/services/MonitoringService.ts (hunk 3 above)
Comment (kajm) [reply source/services/MonitoringService.ts:30]: Fixed

File: source/services/MonitoringService.ts (hunk 4 above)
Comment (kajm) [reply source/services/MonitoringService.ts:10]: moved to separate folder and file

File: source/services/SentryMonitoringService.ts (hunk 5 above)
Comment (kajm) [reply source/services/SentryMonitoringService.ts]: Will keep it for now. Think it could be meaningful if we need to change the sampling rate.

**Review by dannil76 — APPROVED** [review]

**Review by MasterKenth — APPROVED** [review]

**Pull request merged**